review/source/app.py

```diff
@@ -0,0 +1,139 @@
+import tkinter as tk
+from tkinter import ttk
+from tkinter import filedialog, messagebox
+from os import path
+
+
+import source.filesystem as filesystem
+import source.statistics as statistics
+
+class App(tk.Frame):
+    def __init__(self, master=None):
+        super().__init__(master)
+        self.PATH_GABARITO = None
+        self.PATH_RESPOSTAS = None
+        self.PATH_SAVE = None
+
+
+        self.ModuloEstatistica = None
+        
+
+        self.fb_frame = Filebrowser(self).pack(fill='x', expand=True)
+        self.pack(expand=True, fill="x", anchor='n')
+
+        btn = ttk.Button(self, text="Gerar relatorio", command=self.__gerar_relatorio )
+        btn.pack(ipadx=20, ipady=15, pady=5, expand=True, fill="both")
+
+
+    def __gerar_relatorio(self):
+        
+
+        if      self.PATH_GABARITO  \
+            and self.PATH_RESPOSTAS \
+            and self.PATH_SAVE:
+
+            try:
+                gabarito_formatado = filesystem.load_gabarito(self.PATH_GABARITO)
+            except Exception as e:
+                messagebox.showerror("Erro ao abrir arquivo", "Arquivo de gabarito mal formatado")
+                return
+            
+            try:
+                respostas_formatadas = filesystem.load_respostas(self.PATH_RESPOSTAS)
+            except Exception as e:
+                messagebox.showerror("Erro ao abrir arquivo", "Arquivo de respostas mal formatado")
+                return
```

Comment:
Tratamento de erros sempre é muito bem vindo, parabéns! Meu único problema aqui é o fato de você ter usado `Exception`, sendo assim, se ocorrer qualquer outro erro que não seja de "abrir arquivo" ou "formato errado", irá cair nesse caso, o ideal é que você trate cada erro individualmente, se acha que teriam muitos erros, poderia usar o conceito de [Error Handling](https://medium.com/swlh/exceptions-in-python-a545249f1272#id_token=eyJhbGciOiJSUzI1NiIsImtpZCI6IjhjMjdkYjRkMTNmNTRlNjU3ZDI2NWI0NTExMDA4MGI0ODhlYjQzOGEiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJodHRwczovL2FjY291bnRzLmdvb2dsZS5jb20iLCJuYmYiOjE2NzE2NTI5OTAsImF1ZCI6IjIxNjI5NjAzNTgzNC1rMWs2cWUwNjBzMnRwMmEyamFtNGxqZGNtczAwc3R0Zy5hcHBzLmdvb2dsZXVzZXJjb250ZW50LmNvbSIsInN1YiI6IjExMTA3MjU4NDM1MTQ2MjEyODAzNiIsImVtYWlsIjoiZ2FiLnZpYW5hMjAwMUBnbWFpbC5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiYXpwIjoiMjE2Mjk2MDM1ODM0LWsxazZxZTA2MHMydHAyYTJqYW00bGpkY21zMDBzdHRnLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwibmFtZSI6IkdhYnJpZWwgVmlhbmEiLCJwaWN0dXJlIjoiaHR0cHM6Ly9saDMuZ29vZ2xldXNlcmNvbnRlbnQuY29tL2EvQUVkRlRwNzdnMWtNREt3N3I5dTZoOW5SRVNLNTA1TmJidkRiR1JUc2Q2UEZCQT1zOTYtYyIsImdpdmVuX25hbWUiOiJHYWJyaWVsIiwiZmFtaWx5X25hbWUiOiJWaWFuYSIsImlhdCI6MTY3MTY1MzI5MCwiZXhwIjoxNjcxNjU2ODkwLCJqdGkiOiI3YzVkOWZjOGQzZjVjMjlhOTIwNzgzYTk3YTU5Yjk5MGUxMjcwOTRhIn0.PHBus0XB7JHUjhK2q1RBcMgCSr-QnjqR82r1vj5otbVbPgi1r9erproDyIVEFqPZH5SsSHDbyDkm_VBakyr4boUBfsRgMoVnQYA2MbwP3RgEoOFMWrCvvMzqT8vCpjLHKbvGWHd2lY4h-RlYuclxSN5JRDr6yOd9Bfg2AQ2pQ0QWGeX7FlLdqj-KADPSVr0-6cDIzt6DGFxF_3GXUl9_mIvEc9Op4EfZEFWCfZROFp0nzfFlFfuL4a46yOtRW6CzpVT-5fQOKQdKrmhnvv1jX6OLrRzpLS91wP66T4O3bOi8pruwnhrYp00-OwKh_CgB26c92BR0Y6UvtoP0w87F0Q)